### ISSUE TITLE
If silence is close to the song's end it skips next song (sometimes)

### ISSUE THREAD

**Issue by AsVHEn** [kpcee/deadbeef-silence_remover#2]
Hi,
I noticed (because I use a plugin to count number of plays on tracks) undesired behaviour as title says. It seems like _Next song_ action from plugin and from deadbeef doesn't overwrite and duplicates (skiping next full track). I tried to patch it with:

> -    //float length      = deadbeef->pl_get_item_duration( it );
> -    //float pos         = deadbeef->streamer_get_playpos();
> +    float length      = deadbeef->pl_get_item_duration( it );
> +    float pos         = deadbeef->streamer_get_playpos();

> -    else if ( dB_Threshold_Value_End >= 0 && !scan_end_blocked && dB <= dB_Threshold_Value_End && percent > 90.0 )
> +    else if ( dB_Threshold_Value_End >= 0 && !scan_end_blocked && dB <= dB_Threshold_Value_End && percent > 90.0 && length - pos > 0.5)

Thanks for reading.

**Comment by kpcee** [comment]
Yes you are right, this could indeed happen. I have uploaded a new version that should fix the issue.
